x/sync/network_server.go

```diff
@@ -152,48 +152,64 @@ func (s *NetworkServer) HandleChangeProofRequest(
 		return nil // dropping request
 	}
 
-	// override limit if it is greater than maxKeyValuesLimit
+	// override limits if they exceed caps
 	keyLimit := req.KeyLimit
 	if keyLimit > maxKeyValuesLimit {
 		keyLimit = maxKeyValuesLimit
 	}
+
 	bytesLimit := int(req.BytesLimit)
 	if bytesLimit > maxByteSizeLimit {
 		bytesLimit = maxByteSizeLimit
 	}
 
-	// attempt to get a proof within the bytes limit
+	startRoot, err := ids.ToID(req.StartRootHash)
+	if err != nil {
+		return err
+	}
+
+	endRoot, err := ids.ToID(req.EndRootHash)
+	if err != nil {
+		return err
+	}
+
 	for keyLimit > 0 {
-		startRoot, err := ids.ToID(req.StartRootHash)
-		if err != nil {
-			return err
-		}
-		endRoot, err := ids.ToID(req.EndRootHash)
-		if err != nil {
-			return err
-		}
 		changeProof, err := s.db.GetChangeProof(ctx, startRoot, endRoot, req.StartKey, req.EndKey, int(keyLimit))
 		if err != nil {
-			// handle expected errors so clients cannot cause servers to spam warning logs.
-			if errors.Is(err, merkledb.ErrRootIDNotPresent) || errors.Is(err, merkledb.ErrStartRootNotFound) {
-				s.log.Debug(
```

Comment:
The old log was midleading. Just because we had insufficient history doesn't mean the request was invalid.